cmd/runner/main.go

```diff
@@ -44,9 +45,23 @@ func main() {
 	cfg := makeRunnerConfig()
 	fmt.Println(cfg)
 
-	rec := request.Do(cfg)
+	r := requester.New(requester.Options{
+		Concurrency: cfg.RunnerOptions.Concurrency,
+		Requests:    cfg.RunnerOptions.Requests,
+		Duration:    cfg.RunnerOptions.GlobalTimeout,
+		Timeout:     cfg.Request.Timeout,
+	})
 
-	fmt.Println("total:", len(rec))
+	ctx, cancel := context.WithTimeout(context.Background(), cfg.RunnerOptions.GlobalTimeout)
+	defer cancel()
+
+	r.Run(ctx, requester.Target{
+		Method: cfg.Request.Method,
+		URL:    cfg.Request.URL,
+	})
+	report := r.Collect()
```

Comment:
All of that can be hidden behind something like `Requester.RunAndReport(url)`.
The need to extract `ctx` comes from the pipeline nature of `Run` and `Collect`. It may be abstracted inside a single "execute" method.

To discuss.